README.adoc

```diff
@@ -1121,6 +1120,20 @@ users.where(id: ids)
 user.articles.order(:created_at)
 ----
 
+NOTE: When the receiver for `all` is an association, there are methods whose behavior changes by omitting `all`.
+
+The following methods behave differently without `all`:
+
+* delete - https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-delete[with all] / https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-delete[without all]
```

Comment:
I'm torn about whether to add `delete` and `destroy` to this list for the following reasons:

- these methods are not [the targets for verification of cop](https://github.com/rubocop/rubocop-rails/blob/master/lib/rubocop/cop/rails/redundant_active_record_all_method.rb#L39).
- the expected arguments differ depending on whether `all` is present or not. In the first place, it's not possible to omit `all`.
  - even if omitted, unlike `delete_all`, it seems unlikely that one wouldn't notice the change in behavior:thinking: